Tools filter

torresmateo opened the pull request:
This adds a docstring to `get_arcade_tools` that documents and implements the `tools` parameter so not everything in a toolkit is included.

Spartee committed:
Example update

Spartee committed:
Example update 2

Spartee committed:
all

torresmateo committed:
refactor get_arcade_tools to filter with tools parameter

torresmateo committed:
refactor get_arcade_tools to filter with tools parameter

torresmateo committed:
better docstring

EricGustin reviewed (commented):
@Spartee the `get_arcade_tools` does not behave the way that I would expect (irrespective of this current PR).
```py
tools = await get_arcade_tools(
    client, 
    toolkits=["github"], 
    tools=["Google.ListEmails"]
)
# Expected: All github tools + list emails tool
# Actual: All github tools (list emails tool is missing)
```

Seems to me that we are missing a call to `client.tools.formatted.get`. i.e.,
```py
for tool in tools:
    tool_format = await client.tools.formatted.get(
        name=tool, 
        format="openai"
    )
    tool_formats.append(tool_format)
```

This is how [langchain-arcade](https://github.com/ArcadeAI/arcade-ai/blob/2dd77195171baf71f59acbf6f1a3d08ccae3b288/contrib/langchain/langchain_arcade/manager.py#L306) and [crewai-arcade](https://github.com/ArcadeAI/arcade-ai/blob/2dd77195171baf71f59acbf6f1a3d08ccae3b288/contrib/crewai/crewai_arcade/manager.py#L298) behave at least.

torresmateo commented:
Hmm... from the code here I thought the second parameter was a "filter" for the toolkit. I will implement the same behavior as in the other packages for consistency. It's a better DX that way. Making this a draft until I push that!

torresmateo committed:
tools behave like other arcade libraries

sdserranog commented:
This is from the docs, but I’m not sure if we want to change it to match how we do it in other packages.

<img width="788" alt="Screenshot 2025-04-04 at 10 13 51 AM" src="https://github.com/user-attachments/assets/1a16b6e0-6c75-4b5b-b551-f52527e15b28" />

<img width="796" alt="Screenshot 2025-04-04 at 10 12 53 AM" src="https://github.com/user-attachments/assets/843b85e0-3e2c-4e86-b9b1-c29bd0eed228" />

torresmateo commented:
@sdserranog I think we should update the docs to follow changes in this PR, as the API will be consistent with how [langchain-arcade](https://github.com/ArcadeAI/arcade-ai/blob/2dd77195171baf71f59acbf6f1a3d08ccae3b288/contrib/langchain/langchain_arcade/manager.py#L306) and [crewai-arcade](https://github.com/ArcadeAI/arcade-ai/blob/2dd77195171baf71f59acbf6f1a3d08ccae3b288/contrib/crewai/crewai_arcade/manager.py#L298) behave. 

I don't know which order is best to approve these PRs though.

torresmateo committed:
merged docs into tools filter

torresmateo committed:
updated docs to reflect new get_arcade_tools behavior

torresmateo committed:
addressed all of Sergio's comments from the README-update branch

torresmateo commented:
@sdserranog On [this commit](https://github.com/ArcadeAI/openai-agents-arcade/pull/3/commits/a1a1038e1c734163feb250aad84d8aafc8e26782) I addressed all of the comments from the other PR.

Commit:
reviewed all calls to get_arcade_tools

sdserranog reviewed (approved):
I'll go ahead and approve this since the core functionality is working well. We'll still need to clean up/fix the documentation, but we can handle that in a follow-up PR
